src/Parser/DemonHunter/Vengeance/Modules/Talent/SpiritBombSoulsTracked.js

```diff
@@ -21,48 +24,15 @@ class SoulFragments extends Analyzer {
     this.active = this.selectedCombatant.hasTalent(SPELLS.SPIRIT_BOMB_TALENT.id);
   }
 
-  generated = 0;
-  wasted = 0;
-  spent = 0;
-  actual = 0;
-
-  on_byPlayer_cast(event) {
-    const spellId = event.ability.guid;
-    if (spellId === SPELLS.SOUL_FRAGMENT.id) {
-      this.generated += 1;
-      if (this.actual < 5) {
-        this.actual += 1;
-      } else {
-        this.wasted += 1;
-      }
-    }
-  }
-
-  on_byPlayer_removebuffstack(event) {
-    const spellId = event.ability.guid;
-    if (spellId === SPELLS.SOUL_FRAGMENT_STACK.id) {
-      this.spent += 1;
-      this.actual -= 1;
-    }
-  }
-
-  on_byPlayer_removebuff(event) {
-    const spellId = event.ability.guid;
-    if (spellId === SPELLS.SOUL_FRAGMENT_STACK.id) {
-      this.spent += 1;
-      this.actual -= 1;
-    }
-  }
-
   suggestions(when) {
-    const wasterPerGenerated = this.wasted / this.generated;
-    const maximumWaste = Math.floor(0.15 * this.generated);
+    const wasterPerGenerated = this.soulFragmentsTracker.soulsWasted / this.soulFragmentsTracker.soulsGenerated;
+    const maximumWaste = Math.floor(0.15 * this.soulFragmentsTracker.soulsGenerated);
```

Comment:
Looks like 0.15 is a recommendation threshold? Probably should be extracted into a const.

Reply:
Was left over from the person who previously worked on it so i left it, but will clean that up.